### PR TITLE
Improved Windows compatibility for machine command

### PR DESCRIPTION
**Pull request by arixmkii**
Extracting more changes done during experiment in https://github.com/containers/podman/issues/13006 and continuation of https://github.com/containers/podman/pull/15372 (split into independent PRs as the first one is already reviewed and they are pretty independent).

It was checked to compile on MacOS and Windows and as it is purely refactoring, no behavior changes are expected.

##### Fixing command line genarated

The only non platform specific change is 
```
append(cmd, []string{"-qmp", monitor.Network + ":" + monitor.Address.GetPath() + ",server=on,wait=off"}...)
```

This is a correction, because Qemu is using `unix:` (also `tcp:` for TCP) not `unix:/` to set the protocol. It was working before, because address is absolute and additional leading `/` is harmless).

##### Windows aware runtime directory

On windows the only option is to use either user specific or system wide AppData directories

##### Extracting unix specific parts from machine.go

Parts, which only can compile on unix are moved to `_unix` version and `_windows` alternatives (taking into account some limitations, because of lack of signaling) are provided.

#### Does this PR introduce a user-facing change?

```release-note
None
```
[NO NEW TESTS NEEDED]

### CODE REVIEW

File: pkg/machine/qemu/machine_windows.go
Comment (arixmkii) [inline pkg/machine/qemu/machine_windows.go:10]: Will fix it. I believe it is better to add redirects for StdIn, StdOut and StdErr as we only support error codes.

**Comment by arixmkii** [comment]
This doesn't enable Qemu for windows. So, no new functionality added and refactorings should be covered by existent tests, thus no tests provided for now.

**Comment by rhatdan** [comment]
@arixmkii Builds are failing, I added the [NO NEW TESTS NEEDED] flag.

**Comment by mheon** [comment]
/approve
@n1hility PTAL

**Comment by n1hility** [comment]
I have some notes that I will paste shortly, and will post some thoughts on #13006

File: pkg/machine/qemu/machine_windows.go
Comment (n1hility) [inline pkg/machine/qemu/machine_windows.go:34]: If you agree to switch the approach like in the previous comment, I would flip these so that isProcessAlive, just calls the state check method and verifies the return.

**Review by n1hility — CHANGES_REQUESTED** [review]

File: pkg/machine/qemu/machine_windows.go
Comment (arixmkii) [reply pkg/machine/qemu/machine_windows.go]: Change applied and rebased to latest main.

Comment (arixmkii) [reply pkg/machine/qemu/machine_windows.go]: Converted to draft. Need some fixes (export function). Still learning how Go works, because it is my first experience with the language.

Comment (arixmkii) [reply pkg/machine/qemu/machine_windows.go]: Compilation issues fixed.

File: pkg/machine/qemu/machine_windows.go
Comment (n1hility) [inline pkg/machine/qemu/machine_windows.go:7]: Could you move the GetProcessState function to the pkg/machine package (e.g. machine_windows.go) and import pkg/machine instead?

This is where we have shared code already, and the change addresses two benefits:
- Cycles are less likely
- The WSL and Qemu backends are not tied together, and build tags can potentially exclude one or the other.

**Review by n1hility — CHANGES_REQUESTED** [review]
Thanks for the update to the PR! I have one more follow-up (See above)

**Comment by n1hility** [comment]
Also looks like some lingering compile issues

File: pkg/machine/qemu/machine_windows.go
Comment (arixmkii) [reply pkg/machine/qemu/machine_windows.go]: Moved the utility to shared function in machine package.

**Comment by arixmkii** [comment]
I believe a single failed test for rootless ubuntu could be just flaky one or a glitch.

**Comment by n1hility** [comment]
LGTM 

I'll rerun the flake

**Comment by arixmkii** [comment]
@n1hility do you expect some more work with this one (like me rebasing it, etc)? It says it still is waiting for your approval.

**Comment by rhatdan** [comment]
Thanks @arixmkii

**Pull request merged**
